### PR TITLE
fix: transpile server not work

### PR DESCRIPTION
**Pull request by clarkdo**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: packages/config/src/options.js [hunk 1]
```diff
@@ -321,16 +321,19 @@ export function getNuxtConfig(_options) {
...
-        escapeRegExp(
-          path.normalize(
-            module.replace(/\\/g, '/')
-          )
-        )
-      )
-    )
+  const include = []
+  const whitelist = []
+  for (const module of options.build.transpile) {
```
Comment (pi0) [inline packages/config/src/options.js]: Not related to this PR, but module is a CJS reserved keyword :D

File: packages/config/src/options.js [hunk 2]
```diff
@@ -321,16 +321,19 @@ export function getNuxtConfig(_options) {
...
+    if (module instanceof RegExp) {
+      include.push(module)
+      whitelist.push(module)
+    } else {
+      const posixModule = module.replace(/\\/g, '/')
+      include.push(new RegExp(escapeRegExp(path.normalize(posixModule))))
+      whitelist.push(new RegExp(escapeRegExp(posixModule)))
+    }
+  }
+  Object.assign(options.build.transpile, { include, whitelist })
```
Comment (pi0) [inline packages/config/src/options.js]: Is the type of `build.transpile` an object? Also can't we allow users manually providing their `include`/`whitelists`? I feel changing type of an option is little confusing.

File: packages/config/src/options.js (hunk 1 above)
Comment (aldarund) [reply packages/config/src/options.js]: Shouldnt eslint catch this?

File: packages/config/src/options.js (hunk 2 above)
Comment (clarkdo) [reply packages/config/src/options.js]: include/whitelists should be invisible to users, they only config `transpile`.
We adjust them for users.

File: packages/config/src/options.js (hunk 1 above)
Comment (clarkdo) [reply packages/config/src/options.js]: It also happened somewhere else in project, let me change this for now.

Comment (pi0) [reply packages/config/src/options.js]: @aldarund ESLint is not always smart! Try`import path from 'path'` and then modify `path` variable 😆

File: packages/config/src/options.js (hunk 2 above)
Comment (pi0) [reply packages/config/src/options.js]: @clarkdo So if it is an internal resolve mechanism, why don't do it inside `Builder` constructor? Less code execution for `nuxt start` :)

Comment (clarkdo) [reply packages/config/src/options.js]: It could be, let me think

**Review by aldarund — APPROVED** [review]

**Pull request merged**